src/bundles/robot_minigame/helpers/tests.ts

```diff
@@ -0,0 +1,34 @@
+import type { Area, Test } from '../types';
+
+/**
+ * Run the stored tests in state
+ *
+ * @returns if all tests pass
+ */
+export function run_tests({
+  areaLog,
+  tests
+}: {
+  areaLog: Area[],
+  tests: Test[]
+}) : boolean {
+  // Run each test in order
+  for (const test of tests) {
+    // Store status in a variable
+    let success: boolean;
+
+    switch(test.type) {
+      case 'area':
+        success = test.test(areaLog);
+        break;
+      default:
+        success = true;
+    }
```

Comment:
There's only one case, so a ternary suffices and is more readable (plus you get to define `success` as a const:

```suggestion
    // Store status in a variable
    const success = test.type === 'area' ? test.test(areaLog) : true;
```

Of course, representing it this way would make it more obvious that it's essentially a `||` given how short-circuit eval works:

```ts
    // Store status in a variable
    const success = test.type !== 'area' || test.test(areaLog);
```

Though this is less readable so I prefer the first suggestion

Reply:
Wrote it this way for extensibility (as there may be other forms of checks), but I don't mind this suggestion either.